KOGITO-5300 Document existing Kogito Addons (#3888)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/creating-running/chap-kogito-creating-running.adoc

* `rules-quarkus-helloworld`: A Hello World decision service on Quarkus with a single DRL rule unit.
* `ruleunit-quarkus-example` and `ruleunit-springboot-example`: A decision service (on Quarkus or Spring Boot) that uses DRL with rule units to validate a loan application and that exposes REST operations to view application status.
* `dmn-event-driven-quarkus` and `dmn-event-driven-springboot`: A DMN decision service (on Quarkus and Spring Boot) that demonstrates the capability of the {PRODUCT} Event-Driven decisions add-on.

Decision services that use the {PRODUCT} Trusty Service and Explainability Service::
* `dmn-tracing-quarkus`: A DMN decision service on Quarkus that uses the `kogito-addons-quarkus-tracing-decision` add-on to generate tracing events that the {PRODUCT} Trusty Service and Explainability Service can consume and expose.
* `trusty-demonstration`: A tutorial for deploying the `dmn-tracing-quarkus` example application on Kubernetes as a demonstration of {PRODUCT} Trusty Service and Explainability Service capabilities in a cloud environment.

> * `dmn-tracing-quarkus`: A DMN service to evaluate a loan approval and generate tracing events that might be consumed by the Trusty service.

Reviewer: ```suggestion
* `dmn-tracing-quarkus`: A DMN service on Quarkus to evaluate a loan approval and generate tracing events that might be consumed by the Trusty service.
```

* `serverless-workflow-greeting-quarkus` and `serverless-workflow-greeting-springboot`: A Serverless Workflow greeting service with both JSON and YAML workflow definitions.
* `serverless-workflow-service-calls-quarkus` and `serverless-workflow-service-calls-springboot`: A Serverless Workflow service for returning country information.
* `serverless-workflow-events-quarkus`: A Serverless Workflow service for processing job applicant approvals and that showcases event-driven services.

> === {PRODUCT} add-ons

Reviewer: I think we can create the categories as we already have, for example:

Decision services that use add-ons:: (This can consists of dmn-event-driven-quarkus and dmn-event-driven-springboot)
Process services that use add-ons:: this can consists of all the process related add-ons.

You can find multiple examples demonstrating {PRODUCT} add-ons in the https://github.com/kiegroup/kogito-examples[`{PRODUCT_INIT}-examples`] repository in GitHub. A sample of these include:

> * `dmn-event-driven-quarkus`: Demonstrates the capability of the {PRODUCT} Event-Driven Decisions add-on.

Reviewer: I think we can create consistent definitions for the example application. See the following suggestion:
```suggestion
* `dmn-event-driven-quarkus` and `dmn-event-driven-springboot`: A DMN decision service (on Quarkus and Spring Boot) that demonstrates the capability of the {PRODUCT} Event-Driven decisions add-on.
```

* `serverless-workflow-greeting-quarkus` and `serverless-workflow-greeting-springboot`: A Serverless Workflow greeting service with both JSON and YAML workflow definitions.
* `serverless-workflow-service-calls-quarkus` and `serverless-workflow-service-calls-springboot`: A Serverless Workflow service for returning country information.
* `serverless-workflow-events-quarkus`: A Serverless Workflow service for processing job applicant approvals and that showcases event-driven services.

> MongoDB::

Reviewer: ```suggestion
Process services that use MongoDB::
```

> * `process-mongodb-persistence-springboot`: A quickstart project using add-ons that processes deals for travellers using MongoDb. 

Reviewer: ```suggestion
* `process-mongodb-persistence-quarkus` and `process-mongodb-persistence-springboot`: A quickstart project on {Quarkus and Spring Boot) using add-ons that process deals for travelers using MongoDB. 
```

NOTE: The {PRODUCT} Serverless Workflow Spring Boot starter includes the https://github.com/kiegroup/kogito-runtimes/tree/main/springboot/addons/cloudevents[CloudEvents] and https://github.com/kiegroup/kogito-runtimes/tree/main/springboot/addons/events/kafka[Kafka] add-ons to the project for https://github.com/serverlessworkflow/specification/blob/0.6.x/specification.md#Event-Definition[eventing processing].

// Add-Ons

[id="ref-kogito-add-ons_{context}"]
== {PRODUCT} add-ons

> // Purpose statement for the assembly

Reviewer: ```suggestion

```
